packages/financial-templates-lib/src/price-feed/ExpressionPriceFeed.ts

```diff
@@ -117,33 +139,32 @@ class ExpressionPriceFeed extends PriceFeedInterface {
     return this._convertToFixed(this.expressionCode.evaluate(prices), this.getPriceFeedDecimals());
   }
 
-  getPriceFeedDecimals() {
+  public getPriceFeedDecimals(): number {
     return this.decimals;
   }
 
-  async update() {
+  public async update(): Promise<void> {
     // Update all constituent price feeds.
     await Promise.all(Object.values(this.priceFeedMap).map((pf) => pf.update()));
   }
 
   // Takes a BN fixed point number and converts it to a math.bignumber decimal number that the math library can handle.
-  _convertToDecimal(price, inputDecimals) {
+  private _convertToDecimal(price: BN, inputDecimals: number): math.BigNumber {
     const decimals = math.bignumber(inputDecimals);
     const decimalsMultiplier = math.bignumber(10).pow(decimals);
+
     return math.bignumber(price.toString()).div(decimalsMultiplier);
   }
 
   // Takes a math.bignumber OR math.ResultSet of math.bignumbers and converts it to a fixed point number that's
   // expected outside this library. Note: if the price is a ResultSet, the last value is converted and returned.
-  _convertToFixed(price, outputDecimals) {
+  private _convertToFixed(price: math.BigNumber | ResultSet<math.BigNumber>, outputDecimals: number): BN {
     // If the price is a ResultSet with multiple entires, extract the last one and make that the price.
-    if (price.entries) {
+    if ("entries" in price) {
```

Comment:
🙇

Reply:
yeah, typescript doesn't allow you to access a property that isn't guaranteed to exist 🙃